### PR TITLE
added vpktool

### PR DESCRIPTION
**Pull request by trevortomesh**
I wrote a little script for a vpk extractor tool that might help people who aren't coders.

### CODE REVIEW

**Comment by rossengeorgiev** [comment]
Hi @trevortomesh, that's a cool idea!!
Do you think you could develop this further into a more complete CLI tool?
It could use some options and flags. I can think of a few
- -x [path]  extract to directory
- -f --filter  regex filter for file path
- -i --info  print info about the vpk file, version, number of files etc
- -l --list  list files (file size, crc, file path)
- --verify   validate that files are correct
- --no-path   don't create directories, extract files in same directory

**Comment by trevortomesh** [comment]
Sure -- it might take a while as I have my irons in a lot of fires... but, yeah.

**Comment by rossengeorgiev** [comment]
Luckly the modules for this already exist and it's a matter of putting it together. `argparse` will do all the CLI stuff for you, then `vdf` and `regex`, and finally `os` to write stuff to disk in compatible way. Boom! You've got a cross-platform cli tool. Finally, add entry point to the setup file for `vdf`, so that when you install you will get the tool linked in bin and ready for use immediately.

**Comment by trevortomesh** [comment]
I'd be happy to work on it further. Let me know if there's anything you'd want before accepting the pull request. :-)
- I've added some command line arguments -- verbose, inputfile and outputfile. :-)

**Review by rossengeorgiev — DISMISSED** [review]
Need a shebang for python, so it can be executed directly.

`#!/usr/bin/env python`

File: vpktool.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Easy VPK extractor
```
Comment (rossengeorgiev) [inline vpktool.py:3]: Need a shebang for python, so it can be executed directly.

`#!/usr/bin/env python`

**Review by rossengeorgiev — CHANGES_REQUESTED** [review]
Sure that would be nice. (testing github review process)

File: vpktool.py (hunk 1 above)
Comment (trevortomesh) [reply vpktool.py:3]: Done.

**Review by rossengeorgiev — APPROVED** [review]

File: vpktool.py (hunk 1 above)
Comment (trevortomesh) [reply vpktool.py:3]: I think you need to approve the pull request as well.

**Comment by trevortomesh** [comment]
I think you've still got to approve the pull request itself. :-p

**Comment by rossengeorgiev** [comment]
@trevortomesh, I was testing the review functions. The PR is not ready to be merged yet. Do you plan to implement the things I've suggested in my previous comment? Will you be making unittests for vpktool?

**Comment by trevortomesh** [comment]
Of course! I'll have to look into unit test in Python.

**Comment by rossengeorgiev** [comment]
The module already has a test suite, so you can look into that. See the `Makefile` for how to run it. You will want to various functions of the vpktool testable by putting them into seperate functions, so they can be imported and tested. Look at the my bullet point list above first. If you need any help just ask.

**Comment by trevortomesh** [comment]
Great. Will do.

**Comment by rossengeorgiev** [comment]
Closing due to inactivity

**Pull request closed without merging**